ebpf-fw-ebpf/rust-toolchain.toml

```diff
@@ -0,0 +1,13 @@
+[toolchain]
+channel = "nightly"
```

Comment:
```
~/ebpf-firewall]$ nix develop
trace: Rust 1.82.0-nightly-2024-08-07:
Pre-aggregated package `rust` is not encouraged for stable channel since it contains almost all and uncertain components.
Consider use `default` profile like `rust-bin.stable.latest.default` and override it with extensions you need.
See README for more information.
```
Do we actually need `nightly` for anything?

Reply:
Rust-aya uses nightly by default. Should we change it? [https://github.com/aya-rs/aya/blob/3d57d358e40591acf23dfde740697fbfff026410/.github/workflows/ci.yml#L28](url)

Reply:
The link does not work but we don't necessarily have to change it - we just have to have valid reason to have more unstable / floating environment like `nightly`. The context of the link indicates CI which can mean they also build against nightly target but do they support also more stable toolchains? If yes, I propose we depend on that and `rust-aya` can follow `nightly` in their CI.

Reply:
Specifically, they want to use `nightly` builds for eBPF.
[https://github.com/aya-rs/aya/blob/3d57d358e40591acf23dfde740697fbfff026410/xtask/src/public_api.rs#L26](url)

Reply:
ebpf must be compiled with `core` and aya library does not want to use pre-built `core` lib.
[https://github.com/tiiuae/ebpf-firewall/blob/12aeb9e4f69c6b8729abfac66da9534ef008b6c6/xtask/src/build_ebpf.rs#L49](url)

error: the `-Z` flag is only accepted on the nightly channel of Cargo, but this is the `stable` channel
See https://doc.rust-lang.org/book/appendix-07-nightly-rust.html for more information about Rust release channels.
thread 'main' panicked at xtask/src/build_ebpf.rs:65:5:
assertion failed: status.success()
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Reply:
Thanks, that's a good justification and a comment worth having in the `ebpf-firewall` rust-toolchain.toml
That way any future reader sees immediately why nightly is used over stable.